File: Beecrowd/6-Paradigmas/1767-Saco_do_Papai_Noel-5/1767.py
def minkg(maxinum, maxp, size) -> int:
    l=size
    while maxinum[l] == maxp:
        l-=1
    return l+1

def maxprice(price, weight, size, repeat) -> int:

    maxinum=[0 for i in range(0, size+1, +1)]
    maxitens=maxinum[:]
    for i in range(0, repeat):
        value_p=price[i]
        value_w=weight[i]
        for w in range(size, value_w-1, -1):
            if maxinum[w-value_w]+value_p > maxinum[w]:
                maxinum[w]=maxinum[w-value_w]+value_p
                maxitens[w]=maxitens[w-value_w]+1

    toyskg=minkg(maxinum, maxinum[size], size)
    return maxinum[size], maxitens[size], toyskg

for r in range(0, int(input())):
    prices=[]
    weights=[]
    repeat=int(input())
    for i in range(repeat):
        data=input().split(' ')
        prices.append(int(data[0]))
        weights.append(int(data[1]))
    ma, tot, toyskg=maxprice(prices, weights, 50, repeat)
    
    print(ma, "brinquedos")
    print(f"Peso: {toyskg} kg")
    print(f"sobra(m) {repeat-tot} pacote(s)")
    print("")

File: Beecrowd/6-Paradigmas/1767-Saco_do_Papai_Noel-5/1767.cs
using System;

class soluton{

    static int minkg(int[] maxinum, int maxp, int q){
        int l = q;
        while(maxinum[l]==maxp){
            l--;
        }
        return l+1;
    }

    
    static int[] maxprice(int[] price, int[] weight, int repeat, int qtd_weight){
        int[]  maxinum=new int [qtd_weight+1];
        int[] itens=new int[qtd_weight+1];
        int curr_p, curr_w;
        for(int i = 0; i< repeat; i++){
            curr_p=price[i];
            curr_w=weight[i];

            for(int w = qtd_weight; w>curr_w-1; w--){
                if (maxinum[w-curr_w]+curr_p > maxinum[w]){
                    maxinum[w]=maxinum[w-curr_w]+curr_p;
                    itens[w]=itens[w-curr_w]+1;
                }
            }
        }
        int toysk= minkg(maxinum, maxinum[qtd_weight], qtd_weight);
        int[] r={maxinum[qtd_weight], itens[qtd_weight], toysk};
        return r;
        
    }

    static void Main(){

        int tests = int.Parse(Console.ReadLine());
        for(int r=0; r< tests; r++){
            int repeat = int.Parse(Console.ReadLine());
            int[] prices = new int[repeat];
            int[] weights = new int[repeat];
            for (int i=0; i<repeat; i++){
                string[] data = Console.ReadLine().Split(' ');
                prices[i]=int.Parse(data[0]);
                weights[i]=int.Parse(data[1]);
            }
            int r1,r2,r3;
            int[] ra=maxprice(prices, weights, repeat, 50);

            Console.WriteLine(ra[0]+" brinquedos");
            Console.WriteLine("Peso: "+ra[2]+" kg");
            Console.WriteLine("sobra(m) "+(repeat-ra[1])+" pacote(s)");
            Console.WriteLine();
            
        }

    }
}